src/unsavedChildren.js

```diff
@@ -48,8 +50,20 @@ function traverse(
   obj: ParseObject,
   encountered: EncounterMap,
   shouldThrow: boolean,
-  allowDeepUnsaved: boolean
+  allowDeepUnsaved: boolean,
+  counter: number = 0
 ) {
+  counter++;
+
+  if (counter > MAX_RECURSIVE_CALLS) {
+    const message = 'Maximum recursive calls exceeded in traverse function. Potential infinite recursion detected.';
```

Comment:
Could you simply the error message that is thrown into a single sentence that is "less technical"? In case the error message gets returned to the request response as is, it cannot contain any object information for security reasons; instead you may be able to log objects details internally somehow.

Reply:
What are you thinking, something like "Infinite recursion detected in traverse function"?

Also, *cannot contain any object information for security reasons; instead you may be able to log objects details internally somehow* is why I have the `console.error` logging there, that way the server admins (such as myself) can debug what went wrong and see the objects causing these errors. Are you saying you are good with the server-side logging, or that you want it removed?

Reply:
Got it, I missed that you are logging part of that in the console. Maybe something like `Encoding object failed due to high number of recursive calls, likely caused by circular reference within object.`?